lib/c.c

```diff
@@ -370,12 +370,89 @@ int fputc(int c, FILE *stream)
  * FIXME: adopt lite_malloc from musl-libc
  *   https://git.musl-libc.org/cgit/musl/tree/src/malloc/lite_malloc.c
  */
-void *malloc(int size)
+
+typedef struct block_meta {
+    int size;
+    struct block_meta *next;
+    int free;
+} block_meta_t;
+
+block_meta_t *__malloc_global_base;
+block_meta_t *__malloc_global_last;
+
+block_meta_t *get_block_ptr(void *ptr)
 {
-    char *brk = __syscall(__syscall_brk, 0); /* read current break */
-    char *request = __syscall(__syscall_brk, brk + size);
+    return ptr - sizeof(block_meta_t);
+}
+
+block_meta_t *find_free_block(int size)
```

Comment:
Ditto. Inline `find_free_block` function into `main`.

Reply:
Do you means inline `find_free_block` function into `malloc`?

Reply:
Exactly. Please be aware of the fact that the content of `lib/c.c` will be inlined into C source files unless `-no-libc` is specified. Therefore, it is vital to avoid symbol confliction.